maas/resource_maas_volume_group_test.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, []string{"TF_ACC_BLOCK_DEVICE_MACHINE"}) },
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccCheckMAASVolumeGroupDestroy,
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			// Test initial creation
+			{
+				Config: testAccMAASVolumeGroup(machine, name, []string{}, []string{"maas_block_device.bd1.partitions.0.id"}),
+				Check: resource.ComposeTestCheckFunc(append(baseChecks,
+					// we loose about 5MB when creating a volume group, which rounds down to the next GB
+					resource.TestCheckResourceAttr("maas_volume_group.test", "size_gigabytes", "19"),
+					resource.TestCheckResourceAttr("maas_volume_group.test", "block_devices.#", "0"),
+					resource.TestCheckResourceAttr("maas_volume_group.test", "partitions.#", "1"),
+					resource.TestCheckResourceAttrPair("maas_volume_group.test", "partitions.0", "maas_block_device.bd1", "partitions.0.id"),
+				)...),
+			},
+			// Test the update function
+			{
+				Config: testAccMAASVolumeGroup(machine, name, []string{"maas_block_device.bd2.id"}, []string{"maas_block_device.bd1.partitions.0.id"}),
+				Check: resource.ComposeTestCheckFunc(append(baseChecks,
+					resource.TestCheckResourceAttr("maas_volume_group.test", "size_gigabytes", "69"),
+					resource.TestCheckResourceAttr("maas_volume_group.test", "block_devices.#", "1"),
+					resource.TestCheckResourceAttr("maas_volume_group.test", "partitions.#", "1"),
+					resource.TestCheckTypeSetElemAttrPair("maas_volume_group.test", "block_devices.0", "maas_block_device.bd2", "id"),
+					resource.TestCheckResourceAttrPair("maas_volume_group.test", "partitions.0", "maas_block_device.bd1", "partitions.0.id"),
+				)...),
+			},
+			// Test import
+			{
+				ResourceName:      "maas_volume_group.test",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: func(s *terraform.State) (string, error) {
+					rs, ok := s.RootModule().Resources["maas_volume_group.test"]
+					if !ok {
+						return "", fmt.Errorf("resource not found: %s", "maas_volume_group.test")
+					}
+
+					if rs.Primary.ID == "" {
+						return "", fmt.Errorf("resource id not set")
+					}
+					return fmt.Sprintf("%s/%s", rs.Primary.Attributes["machine"], rs.Primary.Attributes["id"]), nil
+				},
+			},
+		},
+	})
+}
+
+func testAccMAASVolumeGroup(machine string, name string, blockDevices []string, partitions []string) string {
+	return fmt.Sprintf(`
+
+data "maas_machine" "machine" {
+  hostname = %q
+}
+
+resource "maas_block_device" "bd1" {
+  machine        = data.maas_machine.machine.id
+  name           = "bd1"
+  size_gigabytes = 25
+  block_size     = 512
+  id_path        = "/dev/bd1"
+  is_boot_device = true
+
+  partitions {
+    size_gigabytes = 20
+  }
+
+  lifecycle {
+    ignore_changes = [
+      partitions[0].fs_type 
```

Comment:
Q: Why do we need this?

Reply:
MAAS automatically creates a filesystem on the boot partition, terraform doesn't like that.

You also can't supply a partition with a filesystem for MAAS to create a volume group, so we can't pre-populate this with the correct value :(

Reply:
Interesting. Would you mind point me to the MAAS logic on the automatic creation of the filesystem?

Reply:
The filesystem type is defined in https://github.com/canonical/maas/blob/master/src/maasserver/enum.py#L576
but i can't find where it's further used